Sem. 03/Solutions/Task02.cpp

```diff
@@ -0,0 +1,217 @@
+#include <iostream>
+#include <fstream>
+
+const char* FILE_NAME = "jobOffers.bin";
+const char* ERROR_MESSAGE = "Error";
+
+const int MAX_NAME_LENGTH = 25;
+
+struct JobOffer {
+    char name[MAX_NAME_LENGTH];
+    int coworkerCount;
+    int vacancyDays;
+    long long salary;
+};
+
+int stringLength(const char* string) {
+    int length = 0;
+
+    for (int i = 0; string[i] != '\0'; ++i) {
+        length++;
+    }
+
+    return length;
+}
+
+bool areStringsEqual(const char* str1, const char* str2) {
+    int length1 = stringLength(str1);
+    int length2 = stringLength(str2);
+
+    if(length1 != length2) {
+        return false;
+    }
+
+    for(int i = 0; i < length1; ++i) {
+
+        if(str1[i] != str2[i]) {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+size_t getFileSize(std::ifstream& in) {
+
+    size_t currentPosition = in.tellg();
+    in.seekg(0, std::ios::end);
+    size_t fileSize = in.tellg();
+    in.seekg(currentPosition, std::ios::beg);
+
+    return fileSize;
+}
+
+namespace jobOfferFunctions {
+
+    void assignName(JobOffer& jobOffer, const char* name) {
+        int length = stringLength(name) + 1;
+
+        for (int i = 0; i < length; ++i) {
+            jobOffer.name[i] = name[i];
+        }
+    }
+
+    void initializeJobOffer(JobOffer& jobOffer, const char* name, const int programmerCount, const int days, const long long salary) {
+        assignName(jobOffer, name);
+        jobOffer.coworkerCount = programmerCount;
+        jobOffer.vacancyDays = days;
+        jobOffer.salary = salary;
+    }
+
+    void writeJobOfferToFile(const size_t arraySize) {
+        JobOffer jobOffer = {};
+        char* name = new char[MAX_NAME_LENGTH];
+        int programmerCount = 0;
+        int days = 0;
+        long long salary = 0;
+
+        std::ofstream out(FILE_NAME, std::ios::out | std::ios::app);
+
+        if(!out.is_open()) {
+            std::cerr << ERROR_MESSAGE;
+            return;
+        }
+
+        for(int i = 0; i < arraySize; ++i) {
+            std::cin.getline(name, MAX_NAME_LENGTH);
+            std::cin >> programmerCount;
+            std::cin >> days;
+            std::cin >> salary;
+            std::cin.ignore();
+
+            initializeJobOffer(jobOffer, name, programmerCount, days, salary);
```

Comment:
instead of calling initializeJobOffer, you can write:
std::cin.getline(jobOffer.name, MAX_NAME_LENGTH);
std::cin >> jobOffer.programmerCount;
and so on...